vizro-core/tests/unit/vizro/models/_controls/test_filter.py

```diff
@@ -83,94 +219,166 @@ def test_set_targets_invalid(self, managers_one_page_two_graphs):
     )
-    def test_set_selector(self, test_input, expected, managers_one_page_two_graphs):
-        filter = vm.Filter(column=test_input)
+    def test_allowed_selectors_per_column_type(self, filtered_column, selector, managers_one_page_two_graphs):
+        filter = vm.Filter(column=filtered_column, selector=selector)
         model_manager["test_page"].controls = [filter]
         filter.pre_build()
-        assert isinstance(filter.selector, expected)
-        assert filter.selector.title == test_input.title()
+        assert isinstance(filter.selector, type(selector))
 
-    @pytest.mark.parametrize("test_input", [vm.Slider(), vm.RangeSlider()])
-    def test_set_slider_values_incompatible_column_type(self, test_input, managers_one_page_two_graphs):
-        filter = vm.Filter(column="country", selector=test_input)
+    @pytest.mark.parametrize(
+        "filtered_column, selector",
+        [
+            ("country", vm.Slider()),
+            ("country", vm.RangeSlider()),
+            ("country", vm.DatePicker()),
+            ("lifeExp", vm.DatePicker()),
+            ("year", vm.Slider()),
+            ("year", vm.RangeSlider()),
+        ],
+    )
+    def test_disallowed_selectors_per_column_type(self, filtered_column, selector, managers_one_page_two_graphs):
+        filter = vm.Filter(column=filtered_column, selector=selector)
         model_manager["test_page"].controls = [filter]
         with pytest.raises(
             ValueError,
-            match=f"Chosen selector {test_input.type} is not compatible with categorical column '{filter.column}'.",
+            match=f"Chosen selector {selector.type} is not compatible with .* column '{filtered_column}'. "
+            f"Allowed selectors for .* column '{filtered_column}' are .*.",
         ):
             filter.pre_build()
 
-    @pytest.mark.parametrize("test_input", [vm.Slider(), vm.RangeSlider()])
-    def test_set_slider_values_shared_column_inconsistent_dtype(
-        self, test_input, managers_shared_column_different_dtype
-    ):
-        filter = vm.Filter(column="shared_column", selector=test_input)
+    @pytest.mark.parametrize(
+        "targets",
+        [
+            ["id_shared_column_numerical", "id_shared_column_temporal"],
+            ["id_shared_column_numerical", "id_shared_column_categorical"],
+            ["id_shared_column_temporal", "id_shared_column_categorical"],
+        ],
+    )
+    def test_set_slider_values_shared_column_inconsistent_dtype(self, targets, managers_shared_column_different_dtype):
+        filter = vm.Filter(column="shared_column", targets=targets)
         model_manager["graphs_with_shared_column"].controls = [filter]
         with pytest.raises(
             ValueError,
-            match=f"Non-numeric values detected in the shared data column '{filter.column}' for targeted charts. "
-            f"Please ensure that the data column contains the same data type across all targeted charts.",
+            match=regex_escape(
```

Comment:
Why is this needed? It just feels a bit weird seeing it imported as `regex_escape` rather than using it as `re.escape` or adding a couple of backslashes.